tool-openssl/x509_test.cc

```diff
@@ -0,0 +1,111 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+#include "openssl/x509.h"
+#include <openssl/err.h>
+#include <gtest/gtest.h>
+#include <stdio.h>
+#include <string>
+#include <vector>
+#include <memory>
+#include "../tool/internal.h"
+#include "internal.h"
+
+X509* CreateAndSignX509Certificate() {
+  X509 *x509 = X509_new();
+  if (!x509) return nullptr;
```

Comment:
Minor nit: Our coding style requires brackets

```suggestion
  if(!x509) {
      return nullptr;
   }
```